### PR TITLE
feat(fe-fpm-writer): support to generate table building block

### PR DESCRIPTION
**Pull request by kranthie-sap**
# Description

* Add support to generate [table building block](https://ui5.sap.com/#/api/sap.fe.macros.Table%23overview)
* Move existing `basic-lrop` integration test app to a nested folder. 
> Note: the `basic-lrop-app` integration test app seems to be broken while previewing in both `js` and `ts` variants
* Add new `custom-page-app` integration test app to test filter bar, chart and table building blocks

## Steps to preview new integration test app
* Run tests in `fe-fpm-package` using the debug environment variable - `UX_DEBUG=true pnpm test`
* Change directory into `test/test-output/integration/custom-page-app/js` or `test/test-output/integration/custom-page-app/ts`
* Run `npm install` followed by `npm run start-mock`
* Hit the Go button
* The application should contain a filter bar, an empty chart (due to missing mock data) and a table with mock data

### CODE REVIEW

File: packages/fe-fpm-writer/test/integration/custom-page-app.test.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                        busy: false,
+                        enableAutoColumnWidth: true,
+                        enableExport: true,
+                        enableFullScreen: true,
+                        enablePaste: true,
+                        filterBar: 'testFilterBar',
+                        header: 'Test Header',
+                        headerVisible: true,
+                        isSearchable: true,
+                        menu: 'testMenuActionKey',
```
Comment (nlunets) [inline packages/fe-fpm-writer/test/integration/custom-page-app.test.ts:89]: I have to check in FE but i don't think menu should be in there (or even what it is), can you please remove it ?

**Review by nlunets — COMMENTED** [review]
Had a look at the generated code, looks good to me, one remark on `menu` property which shouldn't be there in my opinion

**Review by tobiasqueck — CHANGES_REQUESTED** [review]
It looks like something went wrong when you moved the `test-input/integration` folders. The `minUI5Version` is different than in `main` https://github.com/SAP/open-ux-tools/blob/main/packages/fe-fpm-writer/test/test-input/basic-ts/webapp/manifest.json#L59 vs. https://github.com/SAP/open-ux-tools/blob/feat/fe-fpm-writer/add-table-building-block-generator/packages/fe-fpm-writer/test/test-input/integration/basic-lrop-app/ts/webapp/manifest.json#L59

Could you please also rename the `start-mock` scripts to `start` in your new integration test app and add a build script for the ts one. It is not critically important but makes testing the DEBUG apps easier.

**Comment by kranthie-sap** [comment]
> It looks like something went wrong when you moved the test-input/integration folders. The minUI5Version is different than in main https://github.com/SAP/open-ux-tools/blob/main/packages/fe-fpm-writer/test/test-input/basic-ts/webapp/manifest.json#L59 vs. https://github.com/SAP/open-ux-tools/blob/feat/fe-fpm-writer/add-table-building-block-generator/packages/fe-fpm-writer/test/test-input/integration/basic-lrop-app/ts/webapp/manifest.json#L59

@tobiasqueck: Thanks for spotting this. It looks like git merge did something strange while merging changes from https://github.com/SAP/open-ux-tools/pull/1058. Possibly something to do with moving the files.
I reviewed all changes from this PR and made sure they are in my branch.

> Could you please also rename the start-mock scripts to start in your new integration test app and add a build script for the ts one. It is not critically important but makes testing the DEBUG apps easier.

Done.

File: packages/fe-fpm-writer/test/integration/custom-page-app.test.ts (hunk 1 above)
Comment (nlunets) [reply packages/fe-fpm-writer/test/integration/custom-page-app.test.ts:89]: yes it is useless, the doc should be adjusted on our end

File: packages/fe-fpm-writer/test/test-input/integration/custom-page-app/js/webapp/test/locate-reuse-libs.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
+(function (sap) {
```
Comment (tobiasqueck) [inline packages/fe-fpm-writer/test/test-input/integration/custom-page-app/js/webapp/test/locate-reuse-libs.js:1]: We don't need that file in our test folder.

File: packages/fe-fpm-writer/test/integration/custom-page-app.test.ts (hunk 1 above)
Comment (kranthie-sap) [reply packages/fe-fpm-writer/test/integration/custom-page-app.test.ts]: @nlunets: I removed the menu property.

**Review by nlunets — APPROVED** [review]
Looks ok from my point of view.
Checked the code and the output

File: packages/fe-fpm-writer/test/test-input/integration/custom-page-app/js/webapp/test/locate-reuse-libs.js (hunk 2 above)
Comment (tobiasqueck) [reply packages/fe-fpm-writer/test/test-input/integration/custom-page-app/js/webapp/test/locate-reuse-libs.js]: Fixed that for you in https://github.com/SAP/open-ux-tools/pull/1056/commits/3c4194db211c0d2e26eb8ed659ccf1aa72de09bb

**Review by tobiasqueck — APPROVED** [review]
* change is simple and straightforward (you did a very nice job when writing the code for the previous building blocks)
* test coverage is good
* updated test projects works locally as expected
* changeset fits

**Review by nlunets — APPROVED** [review]

**Pull request merged**